game/test/contrib/utils/components/skill/SkillTest.java

```diff
@@ -0,0 +1,62 @@
+package contrib.utils.components.skill;
+
+import static org.junit.Assert.*;
+
+import core.Entity;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.Instant;
+import java.util.function.Consumer;
+
+public class SkillTest {
+
+    private static int value = 0;
+    private final int baseCoolDownInSeconds = 5;
+
+    private Entity entity;
+    private Skill skill;
+    private Consumer<Entity> skillFunction = entity -> value++;
+
+    @Before
+    public void setup() {
+        entity = new Entity();
+        skill = new Skill(skillFunction, baseCoolDownInSeconds);
```

Comment:
Rein theoretisch darf dies nicht in die Setup, da diese beiden Variablen zu 100 % Test abhängig sind und nicht nur die Umgebung um den Test vorbereiten. Bin aber hier nicht mehr auf den aktuellen Stand.

Reply:
> Bin aber hier nicht mehr auf den aktuellen Stand.

Da bist du nicht alleine..

Ich bin eigentlich immer noch Fan von der hier angewendeten Methode. Ich brauch in jedem Test die Entität und den Skill, also why not.... 
Irgendwann müssen wir #213 mal final klären

Reply:
Genau, man braucht in _jedem_ TF die beiden Variablen. Aber wir haben (oder werden haben?) viele TF, die Datei wird unübersichtlich und man sieht bei einem TF weiter unten dann nicht direkt, wie diese beiden Objekte initialisiert sind. Außerdem schliddert man früher oder später in den Fall, dann man im einen TF eine etwas andere Initialisierung haben will also in den anderen ... Eine der Best Practices aus dem in #213 erwähnten Buch war deshalb, in Testfällen (und nur dort) auf DRY zu verzichten und jeden Testfall in sich geschlossen zu schreiben und auf die setup-Methode zu verzichten.

Reply:
@Lena241 wir arbeiten jetzt doch mit @Before und @After etc.
Also nochmal re-re-fixen ^^